client/src/test/java/io/pravega/client/stream/impl/ControllerImplTest.java

```diff
@@ -882,6 +884,30 @@ public void tearDown() {
         testGRPCServer.shutdownNow();
     }
 
+    @Test
+    public void testCredPluginException() throws Exception {
+        // Verify that keep-alive timeout less than permissible by the server results in a failure.
+        NettyChannelBuilder builder = NettyChannelBuilder.forAddress("localhost", serverPort)
+                .keepAliveTime(10, TimeUnit.SECONDS);
+
+        final NettyChannelBuilder channelBuilder;
+        if (testSecure) {
+            channelBuilder = builder.sslContext(GrpcSslContexts.forClient().trustManager(
+                    new File(SecurityConfigDefaults.TLS_CA_CERT_PATH)).build());
+        } else {
+            channelBuilder = builder.usePlaintext();
+        }
+        // Setup mocks.
+        ClientConfig cfg = spy(ClientConfig.builder()
+                .credentials(new DefaultCredentials("pass", "user"))
+                .trustStore(SecurityConfigDefaults.TLS_CA_CERT_PATH)
+                .controllerURI(URI.create((testSecure ? "tls://" : "tcp://") + "localhost:" + serverPort))
+                .build());
+        doThrow(new IllegalStateException("Exception thrown by cred plugin")).when(cfg).getCredentials();
+        ControllerImplConfig controllerCfg = new ControllerImplConfig(1, 1, 1, 1, 1000, cfg);
+             AssertExtensions.assertThrows(IllegalStateException.class, () -> new ControllerImpl(channelBuilder, controllerCfg, this.executor));
```

Comment:
Could we verify that `closeChannel()` of the class under test is invoked when an exception occurs? That'd verify the primary change.

Reply:
I have enabled this verification by enabling `mock-maker-inline` extension of mockito.